src/Common/FitOnFhir.Common/Models/AuthState.cs

```diff
@@ -18,28 +16,28 @@ public AuthState()
         {
         }
 
-        public AuthState(IQueryCollection query)
+        public AuthState(string externalIdentifier, string externalSystem)
```

Comment:
What do you thin about keeping the constructor that takes a IQueryCollection query and just adding another one that takes string externalIdentifier, string externalSystem. It's really up to you, it will remove a few lines of code from the AuthStateService.